packages/block-editor/src/components/block-navigation/style.scss

```diff
@@ -1,244 +1,121 @@
-	.block-editor-block-mover-button {
-		position: relative;
-		width: $button-size;
-		height: $button-size-small;
-
-		// Position the icon
-		svg {
-			position: relative;
-			height: $button-size-small;
+		// Show the block options when any element in
+		// the row is focused.
+		.block-editor-block-navigation-block__options-cell {
+			opacity: 1;
 		}
+	}
 
-		&.is-up-button {
-			align-items: flex-end;
-			svg {
-				bottom: -4px;
-			}
+	&.is-selected,
+	&.is-selected:hover {
+		&,
+		.block-editor-block-navigation-block-contents,
+		.block-editor-block-navigation-block__options-cell .components-button {
+			background: $gray-900;
+			color: $white;
 		}
 
-		&.is-down-button {
-			align-items: flex-start;
-			svg {
-				top: -4px;
-			}
+		.block-editor-block-navigator-descender-line {
+			border-left-color: $white;
 		}
 
-		// Don't show the focus inherited by the Button component.
-		&:focus:enabled {
-			box-shadow: none;
-			outline: none;
+		// Always show the options for the selected block.
+		.block-editor-block-navigation-block__options-cell {
+			opacity: 1;
 		}
+	}
+}
 
-		// Focus style.
-		&:focus {
-			box-shadow: none;
-			outline: none;
-		}
+// The main block button, containing the block icon
+// and label. Pressing this selects the block.
+.block-editor-block-navigation-block-contents {
+	width: 100%;
+	position: relative;
 
-		&:focus::before {
-			@include block-toolbar-button-style__focus();
-		}
+	.block-editor-block-icon {
+		margin-right: $grid-unit-15;
+	}
 
-		// Focus and toggle pseudo elements.
-		&::before {
-			content: "";
-			position: absolute;
-			display: block;
-			border-radius: $radius-block-ui;
-			height: 16px;
-			min-width: 100%;
-
-			// Position the focus rectangle.
-			left: 0;
-			right: 0;
-
-			// Animate in.
-			animation: components-button__appear-animation 0.1s ease;
-			animation-fill-mode: forwards;
-			@include reduce-motion("animation");
-		}
+	&:hover {
+		color: $gray-900;
 	}
+}
 
-	.block-editor-inserter__toggle {
-		background: $gray-900;
-		color: $white;
-		height: $grid-unit-30;
-		margin: 6px 6px 6px 1px;
-		min-width: $grid-unit-30;
+// The block options, including the movers and more menu.
+.block-editor-block-navigation-block__options-cell {
+	position: absolute;
+	top: 0;
+	right: 0;
+}
 
-		&:active {
-			color: $white;
-		}
-	}
+// This is used to indent nested blocks. They stack
+// as nesting increases, creating a larger indentation.
+.block-editor-block-navigator-descender-line {
+	position: relative;
+	flex-shrink: 0;
+	width: 2px;
+	height: 2px;
+	border-left: 2px solid $gray-700;
+	margin: 0 6px 0;
 }
 
+// Hide block number and level description. (i.e. Block 2 of
+// 3, Level 2). This is mostly useful for screen readers.
 .block-editor-block-navigation-block-slot__description,
 .block-editor-block-navigation-block-select-button__description,
 .block-editor-block-navigation-appender__description {
 	display: none;
```

Comment:
We can't use `display: none` if the content is meant to be screen reader accessible. 
Current best practice in this repo is to wrap the screen-reader-only component in [VisuallyHidden](https://github.com/WordPress/gutenberg/blob/abbdb2bfae951bed85c568ece12620bacd1e483f/packages/components/src/visually-hidden/README.md#L1). This ensures it'll have the appropriate styling.